imports/plugins/included/taxes-taxcloud/client/settings/taxcloud.js

```diff
@@ -20,8 +22,23 @@ Template.taxCloudSettings.helpers({
 AutoForm.hooks({
   "taxcloud-update-form": {
     onSuccess: function () {
-      return Alerts.toast(i18next.t("admin.taxSettings.shopTaxMethodsSaved"),
-        "success");
+      if (!TaxCodes.findOne({ taxCodeProvider: "taxes-taxcloud" })) {
+        Meteor.call("taxcloud/getTaxCodes", (err, res) => {
+          if (err) {
+            throw new Meteor.Error(500, "description");
+          } else if (res && Array.isArray(res)) {
+            Alerts.toast(i18next.t("admin.taxSettings.shopTaxMethodsSaved"),
+              "success");
+            res.forEach((code) => {
+              Meteor.call("taxes/insertTaxCodes", Reaction.getShopId(), code, "taxes-taxcloud", (error) => {
+                if (error) {
+                  throw new Meteor.Error("Error populating TaxCodes collection", error);
```

Comment:
This error uses a different format than your previous error.

As noted above, this will not alert the user via the UI, perhaps a Toast alert with the "error" style would be more appropriate here? I'm not too familiar with this tax plugin, so I'll defer to @zenweasel on what exactly the correct process is here.

Reply:
@spencern , thanks for pointing it out, I will fix it